mixins/form.jsx

```diff
@@ -350,15 +353,15 @@ module.exports = {
     }
   },
   signup: function(validate, props) {
-    this.setState({
-      submitting: true
-    });
```

Comment:
This was creating a waiting spinner even if we know it was going to fail. I moved it down to ensure we only start the spinner if basic validation passes.